### PR TITLE
rec: remove the possiblility to disable structured logging

### PR DESCRIPTION
**Pull request by omoerbeek**
### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [X] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by omoerbeek** [comment]
The coverage number are off: the first argument of SLOG is not even compiled in anymore. So there are fewer lines in the executable. Still coveralls seems to count them and thinks the coverage is reduced....

File: pdns/recursordist/rec_control.cc [hunk 1]
```diff
@@ -235,9 +238,51 @@ static RecursorControlChannel::Answer showYAML(const std::string& path)
...
+  }
+
+  if (entry.name) {
+    buf << " subsystem=" << std::quoted(entry.name.get());
+  }
+  buf << " level=" << std::quoted(std::to_string(entry.level));
+  if (entry.d_priority != 0) {
+    buf << " prio=" << std::quoted(Logr::Logger::toString(entry.d_priority));
+  }
+#if 0
```
Comment (Habbie) [inline pdns/recursordist/rec_control.cc:265]: did you mean to leave this block in?

File: pdns/logging.hh [hunk 2]
```diff
@@ -206,8 +206,10 @@ private:
 
 extern std::shared_ptr<Logging::Logger> g_slog;
 
-// Prefer structured logging?
-extern bool g_slogStructured;
+// Prefer structured logging? Since 5.1.0, we always do. We keep a const, to allow for step-by-step
```
Comment (Habbie) [inline pdns/logging.hh:209]: 5.1.0 -> rec 5.1.0

File: pdns/recursordist/pdns_recursor.cc [hunk 3]
```diff
@@ -2666,7 +2658,7 @@ void makeUDPServerSockets(deferredAdd_t& deferredAdds, Logr::log_t log)
       if (address.sin6.sin6_family == AF_INET6 && setsockopt(socketFd, IPPROTO_IPV6, IPV6_V6ONLY, &one, sizeof(one)) < 0) {
         int err = errno;
         SLOG(g_log << Logger::Warning << "Failed to set IPv6 socket to IPv6 only, continuing anyhow: " << stringerror(err) << endl,
-             log->error(Logr::Warning, "Failed to set IPv6 socket to IPv6 only, continuing anyhow"));
+             log->error(Logr::Warning, err, "Failed to set IPv6 socket to IPv6 only, continuing anyhow"));
```
Comment (Habbie) [inline pdns/recursordist/pdns_recursor.cc:2661]: is this meant to just log the errno number?

Comment (omoerbeek) [reply pdns/recursordist/pdns_recursor.cc:2661]: `void Logger::error(Logr::Priority p, int err, const std::string& msg) const` translates `err` using `strerror(err)`

Comment (Habbie) [reply pdns/recursordist/pdns_recursor.cc:2670]: ah, wonderful, I was hoping for something like that

**Review by Habbie — APPROVED** [review]

**Pull request merged**